contracts/BC_fusion/StakeCredit.sol

```diff
@@ -203,24 +229,51 @@ contract StakeCredit is System, Initializable, ReentrancyGuardUpgradeable, ERC20
     }
 
     /**
-     * @return the unbond request at _index and the total length of delegator's unbond queue.
+     * @return the unbond request at _index.
      */
-    function unbondRequest(address delegator, uint256 _index) public view returns (UnbondRequest memory, uint256) {
+    function unbondRequest(address delegator, uint256 _index) public view returns (UnbondRequest memory) {
         bytes32 hash = _unbondRequestsQueue[delegator].at(_index);
-        return (_unbondRequests[hash], _unbondRequestsQueue[delegator].length());
+        return _unbondRequests[hash];
+    }
+
+    /**
+     * @return the total length of delegator's pending unbond queue.
+     */
+    function pendingUnbondRequest(address delegator) public view returns (uint256) {
+        return _unbondRequestsQueue[delegator].length();
     }
 
     /**
-     * @return the total amount of BNB locked in the unbond queue.
+     * @return the total number of delegator's claimable unbond requests.
      */
-    function lockedBNBs(address delegator) public view returns (uint256) {
+    function claimableUnbondRequest(address delegator) public view returns (uint256) {
         uint256 length = _unbondRequestsQueue[delegator].length();
-        if (length == 0) {
-            return 0;
+        uint256 count;
+        for (uint256 i; i < length; ++i) {
+            bytes32 hash = _unbondRequestsQueue[delegator].front();
```

Comment:
you will always get the front item, as you never pop it.

Reply:
Fixed